app/controllers/bids_controller.rb

```diff
@@ -0,0 +1,28 @@
+require 'item_marketplace/presenters/item_presenter'
+require 'bids/presenters/bid_presenter'
+
+class BidsController < ApplicationController
+
+  before_action do
+    check_current_user("You need to be logged in to bid on this item")
+  end
+
+  def create
+    bid = Bid.new(item_id: params[:item_id], bidder_id: current_user.id, amount: params[:amount])
+    Item.transaction do
+      item = Item.lock.find(params[:item_id])
```

Comment:
You never release the lock explicitly here, and I believe ActiveRecord is doing that for you when it closes the connection you're using for this request. For that reason, I might prefer using [with_lock](http://apidock.com/rails/ActiveRecord/Locking/Pessimistic/with_lock) because the block form is more explicit about unlocking outside of the block.